.devcontainer/images/hooks/shared/utils.sh

```
@@ -0,0 +1,369 @@
+
+# Retry function for any command
+# Usage: retry <max_attempts> <delay_seconds> <command...>
+# Example: retry 3 5 curl -o file.tar.gz https://example.com/file.tar.gz
+retry() {
+  local max_attempts=$1
+  local delay=$2
+  shift 2
+  local attempt=1
+  local exit_code=0
+
+  while [ $attempt -le $max_attempts ]; do
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)